packages/xod-client/src/editor/actions.js

```diff
@@ -270,23 +278,30 @@ export const highlightSugessterItem = patchPath => ({
 // Microsoft Edge only supports Text and URL data types
 const getClipboardDataType = () => (isEdge() ? 'Text' : CLIPBOARD_DATA_TYPE);
 
-export const copyEntities = event => (dispatch, getState) => {
-  if (isInput(document.activeElement)) return;
-
-  const state = getState();
-
+// :: State -> Maybe ClipboardEntities
+const getClipboardEntities = (state) => {
   const selection = Selectors.getSelection(state);
-  if (R.isEmpty(selection)) return;
+  if (R.isEmpty(selection)) return Maybe.Nothing();
 
   const currentPatchPath = Selectors.getCurrentPatchPath(state);
-  const currentPatch = R.compose(
-    XP.getPatchByPathUnsafe(currentPatchPath),
+  return R.compose(
+    R.map(currentPatch => getEntitiesToCopy(currentPatch, selection)),
+    XP.getPatchByPath(currentPatchPath),
     ProjectSelectors.getProject
   )(state);
+};
+
+export const copyEntities = event => (dispatch, getState) => {
+  if (isInput(event)) return;
+
+  const state = getState();
 
-  const entities = getEntitiesToCopy(currentPatch, selection);
-  event.clipboardData.setData(getClipboardDataType(), JSON.stringify(entities, null, 2));
-  event.preventDefault();
+  // linter confuses array and a Maybe
```

Comment:
Me too :grimacing: 

I’d use the `foldMaybe(addToClipboard, noop, getClipboardEntities(state))`. Doesn’t matter so much though…